EvtWeightTools/src/EvtWeightRecorder.cc

```diff
@@ -776,6 +775,26 @@ EvtWeightRecorder::record_lheScaleWeight(const LHEInfoReader * const lheInfoRead
   }
 }
 
+void
+EvtWeightRecorder::record_gen_mHH_cosThetaStar(const LHEParticleCollection & lheParticles)
+{
+  std::vector<int> higgs;
+  TLorentzVector h1, h2, H;
+  for (unsigned int indx=0; indx<lheParticles.size(); indx++)
+  {
+    if ( lheParticles[indx].pdgId() ==25 ) higgs.push_back(indx);
+  }
+    if ( higgs.size() ==2 )
```

Comment:
This will only work if the ME is simulated up to Higgs bosons, which are then subsequently decayed by Pythia. In some HH signal samples (bbtautau I think) the Higgs bosons are already decayed at the ME level, which means that there are no status 1 LHE Higgs bosons in the collection, since NanoAOD stores only those by default. Previously, I got around this problem by storing status 2 LHE particles, because we had the luxury of running our own NanoAOD production, but this is not strictly necessary. One could also use the HH decay products to build the 4-momentum of both Higgses, which is why my proposal of storing LHE status 2 particles was rejected by XPOG.

So, using an if statement here would basically mean that computing mHH and cos(theta*) would fail silently for some HH signal samples. My proposal:
1. set `gen_mHH_` and `gen_cosThetaStar_` to some unphysical values in the constructor of this class, eg -1 and -2 would do;
2. throw an exception in `HHWeightInterfaceCouplings::getBinContent` if `TAxis::FindBin` hits an underflow bin (ie return 0). It basically means that whenever `HHWeightInterfaceLO::getWeight`  or `HHWeightInterfaceNLO::getWeight_LOtoNLO` are called with uninitialized mHH and cos(theta*), then it would result in an error;
3. protect the calls to `EvtWeightRecorder::record_gen_mHH_cosThetaStar` with an if statement in `produceNtuple` so that it'd be run only on HH signal samples. At the moment, the function is called for any process and the LHE particles are looped over for every single MC sample, which is simply suboptimal. If the function is called only for HH samples, then it would allow to extend the functionality of building 4-momenta of both Higgs bosons explicitly if they're already decayed at the ME level, and assert that the function really has found two Higgs bosons or their equivalent candidates.

By the way, I don't see `EvtWeightRecorder::record_hhWeight_*` called anywhere. How is this interface supposed to be used in the new FW?

Reply:
One thing is not clear to me, if Higgs are decyed at ME level how to build Higgs boson from the decay product using LHE info. I don't see any branch of mother in LHE branches

Reply:
You can build one Higgs candidate from a SFOS pair of particles that have an invariant mass of 125 GeV (+/- 1%).